backend/backend.go

```diff
@@ -27,6 +29,22 @@ type Backend interface {
 	State() (state.State, error)
 }
 
+// MultiState is an interface that a backend can implement to allow changing
+// between named states depending on the configured environment.
+type MultiState interface {
+	// States returns a list of configured named states and the current state.
+	States() ([]string, string, error)
```

Comment:
The backend shouldn't be responsible for "current" state since that should be local to the user (i.e. with the same remote backend on two different machines one user can point to one env and another can point to another. The backend itself has no concept of "current").

The "current" state should be stored locally in the terraform cache dir (.terraform so it is VCS ignored).